src/App/App.js

```diff
@@ -23,20 +23,22 @@ class App extends Component {
       userName: '',
       userRatings: {ratings: []},
       favorites: [],
+      searchedMovie: ''
     }
 
     this.logOut = this.logOut.bind(this)
     this.showCorrectPage = this.showCorrectPage.bind(this)
     this.toggleButton = this.toggleButton.bind(this)
     this.showMovieDetails = this.showMovieDetails.bind(this)
+    this.updateMovies = this.updateMovies.bind(this)
     this.determineFavoriteStatus = this.determineFavoriteStatus.bind(this)
   }
 
   async componentDidMount() {
     try {
       const data = await getAllMovies()
       const favorites = await getFavorites()
-      this.setState({movies: data.movies, favorites: favorites})
+      this.setState({movies: data.movies, permanentMovies: data.movies, favorites: favorites})
```

Comment:
I'm a little confused by `permanentMovies` in state! It looks like. It looks like it's the same thing as `movies`... I could totally be missing something though...

Reply:
Yeah I should have put a note on that. It's for the search ability. the page automatic re-renders the cards based on each letter typed into the search input. 
When they hit the back space (or delete the input value) then the page can re-render with permanent movies instead of trying to re-render with movies, which will have nothing in it after it runs out of movies that match the search. 
Hopefully thats not too confusing. I can definitely show you what I mean on a call too.